aten/src/ATen/mps/MPSGeneratorImpl.h

```diff
@@ -0,0 +1,52 @@
+//  Copyright © 2022 Apple Inc.
+
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <ATen/core/PhiloxRNGEngine.h>
+#include <c10/core/GeneratorImpl.h>
+#include <c10/util/Optional.h>
+
+namespace at {
+namespace mps {
+namespace detail {
+
+static const uint32_t PHILOX_STATE_N = 7;
+struct rng_data_pod {
+  std::array<uint32_t, PHILOX_STATE_N> state{1};
+  uint64_t seed = default_rng_seed_val;
+};
+
+TORCH_API const Generator& getDefaultMPSGenerator();
+TORCH_API Generator createMPSGenerator(uint64_t seed_val = default_rng_seed_val);
+
+} // namespace detail
+} // namespace mps
+
+struct TORCH_API MPSGeneratorImpl : public c10::GeneratorImpl {
+  // Constructors
+  MPSGeneratorImpl(uint64_t seed_in = default_rng_seed_val);
```

Comment:
Do you need the default value here?

Reply:
The code is based on the existing `CPUGeneratorImpl.h`:
```
struct TORCH_API CPUGeneratorImpl : public c10::GeneratorImpl {
  CPUGeneratorImpl(uint64_t seed_in = default_rng_seed_val);
...
createCPUGenerator(uint64_t seed_val = default_rng_seed_val);
```

It's just there to set a default value when we create a Generator's instance implicitly without calling the createCPUGenerator().